src/java.base/share/native/libjli/java.c

```diff
@@ -1209,7 +1209,7 @@ ParseArguments(int *pargc, char ***pargv,
 
     *pret = 0;
 
-    while ((arg = *argv) != 0 && *arg == '-') {
+    while (argc > 0 && *(arg = *argv) == '-') {
```

Comment:
AFAICS this loop terminates at line 1388 and nowhere in the loop body does argc get modified. ??

Reply:
Reference to argc is passed down to GetOpt, where it is updated.

Reply:
Ah I see - ouch!